azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/AvailabilitySet.java

```diff
@@ -64,27 +64,42 @@
 
 
     /**************************************************************
-     * Fluent interfaces to provision an AvailabilitySet
+     * Fluent interfaces to create an AvailabilitySet
      **************************************************************/
 
     /**
+     * Container interface for all the definitions
+     */
+    public interface Definitions extends
+        DefinitionBlank,
+        DefinitionWithGroup,
+        DefinitionAfterGroup,
+        DefinitionCreatable {
+    }
+    
+    /**
      * The first stage of an availability set definition
      */
-    interface DefinitionBlank extends GroupableResource.DefinitionWithRegion<DefinitionWithGroup> {
+    public interface DefinitionBlank extends GroupableResource.DefinitionWithRegion<DefinitionWithGroup> {
```

Comment:
yes, we want `public` for inner interfaces so that it's visible outside the package. For example without `PublicIpAddress.DefinitionCreatable` being public we cannot define below method in `VirtualMachine`

```java
interface DefinitionWithPublicIP {
    DefinitionWithOS withNewPublicIP(PublicIpAddress.DefinitionCreatable creatable);
    ...
}
```